wgsl/index.bs

```diff
@@ -534,7 +534,196 @@ When unclear from context, this specification indicates
+
+Triggered diagnostics [=behavioral requirement|will=] be processed as follows:
+1. For each diagnostic *D*, find the [=diagnostic filter=]
+    with the smallest [=affected range=] that contains D's [=diagnostic/triggering location=], and which has the same [=diagnostic/triggering rule=].
+    * If such a filter exists, apply it to *D*, updating *D*'s [=diagnostic/severity=].
+    * Otherwise *D* remains unchanged.
+2. Discard diagnostics that have severity [=severity/off=].
+3. If at least one remaining diagnostic *DI* has severity [=severity/info=], then:
+    * Other [=severity/info=] diagnostics with same triggering rule *may* be discarded, leaving only the original diagnostic *DI*.
+4. If at least one remaining diagnostic *DW* has severity [=severity/warning=], then:
+    * Other [=severity/info=] or [=severity/warning=] diagnostics with same triggering rule *may* be discarded, leaving only the original diagnostic *DW*.
+5. If at least one remaining diagnostic has [=severity/error=] severity, then:
+    * Other diagnostics *may* be discarded, including other diagnostics with [=severity/error=] severity.
+    * A [=program error=] is generated.
+         * The error is a [=shader-creation error=] if the diagnostic was triggered at [=shader module creation=] time.
+         * The error is a [=pipeline-creation error=] if the diagnostic was triggered at [=pipeline creation=] time.
+6. If processing during [=shader module creation=] time, the remaining diagnostics populate the
+    [[WebgPU#dom-gpucompilationinfo-messages|messages]] member of the
+    [[WebGPU#gpucompilationinfo|WebGPU GPUCompilationInfo]] object.
+7. If processing during [=pipeline creation=], the [=severity/error=] diagnostics are reported
+    in the nearest enclosing [[WebGPU#gpu-error-scope|WebGPU GPU error scope]].
+    Issue: TODO: Adjust for [#3682](https://github.com/gpuweb/gpuweb/issues/3682).
+
+Note: The rules allow an implementation to stop processing a WGSL program as soon as an error is detected.
+Additionally, an analysis for a particular kind of warning can stop on the first warning,
+and an analysis for a particular kind of info diagnostic can stop on the first occurrence.
+
+### Filterable Triggering Rules ### {#filterable-triggering-rules}
+
+Most [=diagnostics=] are unconditionally reported to the WebGPU application.
+Some kinds of diagnostics can be [[#diagnostic-filtering|filtered]], in part by naming their [=diagnostic/triggering rule=].
+The following table lists the standard set of triggering rules that can be filtered.
+
+<table class='data'>
+  <caption>Filterable diagnostic triggering rules</caption>
+  <thead>
+    <tr><th>Filterable Triggering Rule<th>Default Severity<th>Triggering Location<th>Description
+  </thead>
+
+  <tr>
+    <td><dfn noexport dfn-for="trigger">derivative_uniformity</dfn>
+    <td>[=severity/error=]
+    <td>A call to a [=builtin functions that compute a derivative|builtin function that computes a derivative=].
```

Comment:
Since we're defining the triggering location, shouldn't this spell out exactly what the location is, in terms of the grammar?

Reply:
I've updated this to say "call site", and then refined "call site" itself to say it's the location of the first token in the parsed call_phrase.
Perhaps call site should really be the range of source locations across the entire call_phrase.  It would work either way but then we'd be talking about source ranges all the time.